src/mango/test/16-index-selectors-test.py

```diff
@@ -189,6 +189,7 @@ def test_uses_partial_index_with_non_indexable_selector(self):
 @unittest.skipUnless(mango.has_text_service(), "requires text service")
 class IndexSelectorText(mango.DbPerClass):
     def setUp(self):
+        self.db = mango.Database(f"{mango.random_db_name()}_{str(self).split()[0]}")
         self.db.recreate()
```

Comment:
There was [recent work](https://github.com/apache/couchdb/commit/eb6a74d5603d4bac1d77fc895946ffbdd99f4b97) done on the `recreate` function, but it didn't seem to address the raciness of reusing db names.

Reply:
Thank you for the link!

Reply:
Yes, that scope of that commit was only limited to solving the infinite loop of retries with recursion.